.github/workflows/lint-test.yaml

```diff
@@ -1,7 +1,15 @@
+# See https://github.com/helm/chart-testing-action
 name: Lint and Test Charts
 
-on: [pull_request, workflow_dispatch]
+on: 
+  pull_request:
+    paths:
+      - 'charts/**/ci/**'
+      - 'charts/**/templates/**'
+      - 'charts/**/Chart.yaml'
+      - 'charts/**/values.yaml'
```

Comment:
@peterzhuamazon I noticed you asked why these files are special. 

The Helm chart-testing action performs version checking, YAML schema validation on 'Chart.yaml', YAML linting on 'Chart.yaml' and 'values.yaml', and maintainer validation [Ref](https://github.com/helm/chart-testing/blob/main/doc/ct_lint.md#synopsis). 

Any other modifications in this repository [SHOULD NOT](https://datatracker.ietf.org/doc/html/rfc2119) trigger linting and testing as they aren't modifying the charts.

Reply:
Could you explain a bit more exactly what issue this is addressing? Why do the `ct` defaults not work?

Reply:
@smlx I am trying to avoid linting from throwing errors for changes that do not touch the charts but `ct list-changed` doesn't seem to work as I'd expect.

[Here](https://github.com/opensearch-project/helm-charts/runs/3775918231?check_suite_focus=true#step:7:18) is the failure in this repository when I don't add paths on the trigger. It's just weird as `ct list-changed` returns both charts and I did not change any chart-related values that `ct` cares about:

```shell
ct list-changed --config ct.yaml

charts/opensearch-dashboards
charts/opensearch
```

Here's the diff between this branch and `main`.

<details>
<summary>diff</summary>

<p>
```diff
diff --git a/.github/workflows/lint-test.yaml b/.github/workflows/lint-test.yaml
index 0ac454f..2f7237f 100644
--- a/.github/workflows/lint-test.yaml
+++ b/.github/workflows/lint-test.yaml
@@ -1,7 +1,8 @@
+# See https://github.com/helm/chart-testing-action
 name: Lint and Test Charts
 
 on: [pull_request, workflow_dispatch]
-  
+
 jobs:
   lint-test:
     runs-on: ubuntu-latest
diff --git a/README.md b/README.md
index 25c7068..fad0e2d 100644
--- a/README.md
+++ b/README.md
@@ -16,22 +16,28 @@ A community repository for Helm Charts of OpenSearch Project.
 
 ## Status
 
-![Testing](https://github.com/opensearch-project/helm-charts/workflows/Lint%20and%20Test%20Charts/badge.svg)
-![Release](https://github.com/opensearch-project/helm-charts/workflows/Release%20Charts/badge.svg)
+[![Lint and Test Charts](https://github.com/opensearch-project/helm-charts/actions/workflows/lint-test.yaml/badge.svg)](https://github.com/opensearch-project/helm-charts/actions/workflows/lint-test.yaml)
+ [![Release Charts](https://github.com/opensearch-project/helm-charts/actions/workflows/release.yaml/badge.svg)](https://github.com/opensearch-project/helm-charts/actions/workflows/release.yaml)
 
 ## Installation
 
+To install the OpenSearch Helm charts, execute the following commands:
+
 ```shell
 helm repo add opensearch https://opensearch-project.github.io/helm-charts/
 helm repo update
 ```
 
-You can then run `helm search repo opensearch` to see the charts.
+Once the charts repository reference is added, you can run the following command to see the charts.
+
+```shell
+helm search repo opensearch
+```
 
 ## Change Logs
 
 Please review the [OpenSearch](charts/opensearch/CHANGELOG.md) and the
-[OpenSearch Dashboards](charts/opensearch/CHANGELOG.md) change logs for the latest 
+[OpenSearch Dashboards](charts/opensearch-dashboards/CHANGELOG.md) change logs for the latest 
 release details.
 
 ## Contributing
diff --git a/charts/opensearch-dashboards/CHANGELOG.md b/charts/opensearch-dashboards/CHANGELOG.md
index 8128fc7..031c7f8 100644
--- a/charts/opensearch-dashboards/CHANGELOG.md
+++ b/charts/opensearch-dashboards/CHANGELOG.md
@@ -27,5 +27,5 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 ### Security
 
-[Unreleasedd]: https://github.com/opensearch-project/helm-charts/compare/opensearch-dashboards-1.0.2...HEAD
-[opensearch-dashboards-1.0.2]: https://github.com/opensearch-project/helm-charts/compare/opensearch-dashboards-1.0.1...opensearch-dashboards-1.0.2
+[Unreleased]: https://github.com/opensearch-project/helm-charts/compare/opensearch-dashboards-1.0.2...HEAD
+[1.0.2]: https://github.com/opensearch-project/helm-charts/compare/opensearch-dashboards-1.0.1...opensearch-dashboards-1.0.2
diff --git a/charts/opensearch-dashboards/README.md b/charts/opensearch-dashboards/README.md
new file mode 100644
index 0000000..19af1a9
--- /dev/null
+++ b/charts/opensearch-dashboards/README.md
@@ -0,0 +1,34 @@
+# OpenSearch Dashboards Helm Chart
+
+ This Helm chart installs [OpenSearch](https://github.com/opensearch-project/OpenSearch) with configurable TLS, RBAC and much more configurations. This chart caters a number of different use cases and setups.
+
+ - [Requirements](#requirements)
+ - [Installing](#installing)
+ - [Uninstalling](#uninstalling)
+
+ ## Requirements
+
+ * Kubernetes >= 1.14
+ * Helm >= 2.17.0
+ * We recommend you to have 8 GiB of memory available for this deployment, or at least 4 GiB for the minimum requirement. Else, the deployment is expected to fail.
+
+ ## Installing
+
+ Once you've added this Helm repository as per the repository-level [README](../../README.md#installing)
+ then you can install the chart as follows:
+
+ ```shell
+ helm install my-release opensearch/opensearch-dashboards
+```
+
+ The command deploys OpenSearch Dashboards with its associated components on the Kubernetes cluster in the default configuration.
+
+ **NOTE:** If using Helm 2 then you'll need to add the [`--name`](https://v2.helm.sh/docs/helm/#options-21) command line argument. If unspecified, Helm 2 will autogenerate a name for you.
+
+ ## Uninstalling
+ To delete/uninstall the chart with the release name `my-release`:
+
+ ```shell
+ helm uninstall my-release
+ ```
+ 
\ No newline at end of file
diff --git a/charts/opensearch/README.md b/charts/opensearch/README.md
index 728145f..a588062 100644
--- a/charts/opensearch/README.md
+++ b/charts/opensearch/README.md
@@ -3,8 +3,9 @@
 This Helm chart installs [OpenSearch](https://github.com/opensearch-project/OpenSearch) with configurable TLS, RBAC and much more configurations. This chart caters a number of different use cases and setups.
 
 - [Requirements](#requirements)
-- [Installing the chart](#installing)
-- [Uninstalling the chart](#uninstalling)
+- [Installing](#installing)
+- [Uninstalling](#uninstalling)
+- [Usage Notes](#usage-notes)
 - [Configuration](#configuration)
 
 ## Requirements
@@ -15,22 +16,40 @@ This Helm chart installs [OpenSearch](https://github.com/opensearch-project/Open
 
 ## Installing
 
-To install the chart with the release name `my-release`:
-- Switch to opensearch directly after cloning the repo
-  `❯ cd charts/opensearch`
-- Run `❯ helm package .`
-- Install using Helm 3:
-`❯ helm install my-release opensearch-1.0.0.tgz`
--  Install using Helm 2
-`❯ helm install --name my-release opensearch-1.0.0.tgz`
+Once you've added this Helm repository as per the repository-level [README](../../README.md#installing) then you can install the chart as follows:
+
+ ```shell
+ helm install my-release opensearch/opensearch
+ ```
 
 The command deploys OpenSearch with its associated components (data statefulsets, masters, clients) on the Kubernetes cluster in the default configuration.
 
+**NOTE:** If using Helm 2 then you'll need to add the [`--name`](https://v2.helm.sh/docs/helm/#options-21) command line argument. If unspecified, Helm 2 will autogenerate a name for you.
+
 ## Uninstalling
 To delete/uninstall the chart with the release name `my-release`:
+
+```shell
+helm uninstall my-release
 ```
-❯ helm uninstall my-release
-```
+
+## Usage Notes
+
+* The chart deploys a StatefulSet and by default will do an automated rolling
+  update of your cluster. It does this by waiting for the cluster health to become
+  green after each instance is updated.
+* It is important to verify that the JVM heap size in `opensearchJavaOpts` and
+  to set the CPU/Memory `resources` to something suitable for your cluster.
+* To simplify chart and maintenance each set of node groups is deployed as a
+  separate Helm release. Without doing this it isn't possible to resize persistent
+  volumes in a StatefulSet. By setting it up this way it makes it possible to add
+  more nodes with a new storage size then drain the old ones. It also solves the
+  problem of allowing the user to determine which node groups to update first when
+  doing upgrades or changes.
+* We have designed this chart to be very un-opinionated about how to configure
+  OpenSearch. It exposes ways to set environment variables and mount secrets
+  inside of the container. Doing this makes it much easier for this chart to
+  support multiple versions with minimal changes.
 
 ## Configuration
 
diff --git a/ct.yaml b/ct.yaml
index 8554b40..5f24888 100644
--- a/ct.yaml
+++ b/ct.yaml
@@ -1,4 +1,3 @@
-# See https://github.com/helm/chart-testing-action
 # See https://github.com/helm/chart-testing#configuration
 target-branch: main
 helm-extra-args: --timeout 800s
```

</p>
</details>

Based on my testing, it seems that if we change **any** file within the target charts folders then `ct list-changed` "thinks" the chart has changed and it errors indicating we need a version bump.

You can see in [this](https://github.com/mprimeaux/helm-charts/commit/1035b34e673b4b5fabd94d1a53fd103aff7c8a0b) commit I modified the root-level `ADMINS.md` file and `ct list-changed` returned no changes resulting in [success](https://github.com/mprimeaux/helm-charts/actions/runs/1298050846).

You can see in [this](https://github.com/mprimeaux/helm-charts/commit/9d83bead84848196acf1bdb9c67abd43dde1c5e0) commit I modified the `CHANGELOG.md` file in the `opensearch` chart directory and `ct list-changed` returned changes resulting in [failure](https://github.com/mprimeaux/helm-charts/runs/3775961477?check_suite_focus=true#step:7:17).

Obviously, this is not the behavior we seek as we want the freedom to change files unrelated to the chart without requiring a version bump.

I haven't debugged this yet but it seems the [`ListChangedFilesInDirs`](https://github.com/helm/chart-testing/blob/fdab4b022098cf2f1fa0e92edeae4698b726f7af/pkg/tool/git.go#L60) function in `ct` isn't filtering relevant chart changes but is instead "blindly" looking for *any* changes in the chart folder.

All this said, I believe the paths are needed.

Reply:
> Obviously, this is not the behavior we seek as we want the freedom to change files unrelated to the charts without requiring a version bump.

I disagree with this. Documentation is just as important as templates, and chart contents should be immutably associated with unique version numbers. All PRs which make changes in the chart subdirectory should be both version bumped and released.

The Helm tooling is opinionated for a reason: this policy is what the collective experience of the Helm community has found to be best practice. Lets stick to the defaults as much as we possibly can.

Reply:
@smlx I've removed the `paths:` stanza. Please review and let me know.